src/plcc/load_spec/validate_spec/validate_lexical_spec_test.py

```diff
@@ -0,0 +1,151 @@
+from pytest import raises, mark, fixture
+
+from ..load_rough_spec.parse_lines import Line
+from ..load_rough_spec.parse_blocks import Block
+from .validate_lexical_spec import ValidationError, validate_lexical_spec
+from ..parse_spec.parse_lexical_spec import LexicalRule, LexicalSpec
+
+def test_empty_no_errors():
+    lexicalSpec = makeLexicalSpec([])
+    errors = validate_lexical_spec(lexicalSpec)
+    assert len(errors) == 0
+
+def test_None_no_errors():
+    lexicalSpec = makeLexicalSpec(None)
+    errors = validate_lexical_spec(lexicalSpec)
+    assert len(errors) == 0
+
+def test_lowercase_name_format_error():
+    invalidName = makeLexicalRule(makeLine("invalid_name \'n\'"), False, "invalid_name", "")
```

Comment:
This is hard to read because it's not clear what all the parameters are for makeLexicalRule, and how they relate to the test.

What if it could look like this...

```
invalidName = makeSpecWithOneTokenRule(name='invalid_name')
errors = validate_lexical_spec(lexicalSpec)
assert len(errors) == 1
assert errors[0] == makeInvalidNameFormatError(invalidName)
```

Better still. You have so many "invalid name" tests. How about making a method that does all of this... then the body of each test is...

```
assertInvalidName('invalid_name')
```

Then you'd have a bunch of one-line tests for invalid names.

After you do this, take another look at other tests and ask yourself, "Do they talk about things that are not relevant to the test?" and "Is this test very similar to another test?" If you answer yes to either, try to fix it.